### PR TITLE
Add subprocess

### PR DESCRIPTION
**Pull request by palaviv**
In CPython subprocess is actually a pure python lib that uses `os` functions but I think the [subprocess](https://docs.rs/subprocess/0.1.18/subprocess/index.html) library in rust fits better here.

### CODE REVIEW

**Comment by windelbouwman** [comment]
W00t, this library was modeled after python's API. This is nice!

File: vm/src/stdlib/subprocess.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        },
+        None => Ok(subprocess::Redirection::None),
+    }
+}
+
+fn convert_to_file_io(file: &Option<File>, mode: String, vm: &VirtualMachine) -> PyResult {
+    match file {
+        Some(ref stdin) => io_open(
+            vm,
+            PyFuncArgs::new(
```
Comment (windelbouwman) [inline vm/src/stdlib/subprocess.rs]: There is a `From` implementation for `PyFuncArgs` for `Vec`, so you might be able to reduce this to:
```rust
io_open(
            vm,
            vec![
                    vm.new_int(raw_file_number(stdin.try_clone().unwrap())),
                    vm.new_str(mode),
            ],
)
```

**Comment by windelbouwman** [comment]
@palaviv this looks good to me, except for one minor comment. Please feel free to merge this.

File: vm/src/stdlib/subprocess.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/stdlib/subprocess.rs]: Did you mean to append a `.into()` on the `vec!` call?

Comment (windelbouwman) [reply vm/src/stdlib/subprocess.rs]: Yes, you are right. I was confused with the `invoke` function which accepts an `Into<PyFuncArgs>` type.

**Pull request merged**